### PR TITLE
#164139647 Create a page to view and edit profile

### PR DESCRIPTION
**Pull request by Dubby20**
#### What does this PR do?
- It consumes an API endpoint that gets and edits a user profile.

#### Description of Task to be completed?
- Creates a single page that displays and edits the user's profile and statistics

#### How should this be manually tested?
- By visiting this url `localhost:3000:userprofile` and `localhost:3000:editprofile`

#### Any background context you want to provide?
- None

#### What are the relevant pivotal tracker stories?
[#164139647](https://www.pivotaltracker.com/story/show/164139647)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A

### CODE REVIEW

File: src/views/profilePage/EditProfile.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    editProfile(formData);
+    props.history.push('/userprofile');
+  };
+
+  return (
+    <div className="profile-container">
+      <Container>
+        <Row>
+          <Col md={3}>
+            <Image className="profile-image" src="https://res.cloudinary.com/djdsxql5q/image/upload/v1547914942/ireporter/ujz7ozprk8hkyszjju7d.jpg" />
```
Comment (codinger41) [inline src/views/profilePage/EditProfile.jsx]: @Dubby20 let's save this image URL in a variable to make the code look cleaner

File: src/views/profilePage/ViewProfile.jsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  const { getProfileRequest: getUserProfile, profile } = props;
+  useEffect(() => {
+    getUserProfile();
+  }, []);
+  return (
+    <div className="profile-container">
+      <Container>
+        <Row>
+          <Col md={3}>
+            <Image className="profile-image" src="https://res.cloudinary.com/djdsxql5q/image/upload/v1547914942/ireporter/ujz7ozprk8hkyszjju7d.jpg" />
```
Comment (codinger41) [inline src/views/profilePage/ViewProfile.jsx]: @Dubby20 same with this image URL

File: src/assets/default.scss [hunk 3]
```diff
@@ -17,6 +17,9 @@ $primary-color: #999;
 $box-shadow-color: 0 6px 20px 1px rgba(0, 0, 0, .15);
 $white-color: #fff;
 $yellow: #e2d754;
+$deep-purple: hsl(279, 68%, 19%);
+$deep-purple-grad: rgba(60, 16, 83, .9);
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/assets/default.scss]: Add leading zeros to this and subsequent. The sass lint rules have been updated

File: src/assets/default.scss [hunk 4]
```diff
@@ -28,10 +31,104 @@ $hover-background: rgb(172, 163, 64);
...
+
+a,
+.yellow-button {
+  transition: ease-in-out .4s;
+}
+
+.purple-gradient-bg {
+  background-attachment: fixed;
+  background-image: linear-gradient(to right, $deep-purple-grad, $light-purple-grad),
+  url('https://res.cloudinary.com/jessam/image/upload/v1554725326/tachina-lee-42980-unsplash_1.jpg');
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/assets/default.scss]: Consider adding a rule escape line to this.

File: src/assets/default.scss [hunk 5]
```diff
@@ -28,13 +31,108 @@ $hover-background: rgb(172, 163, 64);
...
+    color: $white;
+  }
+
+}
+
+.container-social-login {
+  margin-bottom: 3rem;
+  margin-top: 3rem;
+
+  & .info {
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/assets/default.scss]: Consider removing the &, the styling should still work

File: src/components/articleSummary/articleSummary.scss [hunk 6]
```diff
@@ -1,3 +1,14 @@
+@import '../../assets/default';
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/components/articleSummary/articleSummary.scss:1]: Try if double quoted will work.

File: src/redux/reducers/authReducer.js [hunk 7]
```diff
@@ -1,8 +1,11 @@
-import { LOGIN, LOGOUT, CURRENT_USER } from '../actions/actionTypes';
+import {
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/redux/reducers/authReducer.js:1]: Try adding a line to this.

File: src/views/Article/article.scss [hunk 8]
```diff
@@ -1,14 +1,14 @@
 .edit-delete-container {
-  border: 1px solid black;
-  width: 80%;
-  margin: auto;
+  border: 1px solid #000000;
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/views/Article/article.scss]: Let's see if we can move this to default.scss

File: src/views/profilePage/profile.scss [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+.avatar-btn {
+  background-color: $yellow-color !important;
+  border: 0 !important;
+  border-color: $yellow-color;
+  border-radius: 0 !important;
+  color: $black-color !important;
+  margin-top: 1.25rem;
+  outline-style: none !important;
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/views/profilePage/profile.scss]: Let's put this question to the team.

**Review by olajide1234 — COMMENTED** [review]
Thanks for coming through with this PR Jacinta, you are an absolute rockstar. Let's fix these few comments, please.

File: src/components/articleSummary/articleSummary.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (Dubby20) [reply src/components/articleSummary/articleSummary.scss:1]: Thanks. But eslint allows only single quotes.

File: src/views/profilePage/EditProfile.jsx (hunk 1 above)
Comment (Dubby20) [reply src/views/profilePage/EditProfile.jsx]: Alright Thanks.

File: src/views/profilePage/ViewProfile.jsx (hunk 2 above)
Comment (Dubby20) [reply src/views/profilePage/ViewProfile.jsx]: Done

File: src/assets/default.scss [hunk 10]
```diff
@@ -28,13 +31,108 @@ $hover-background: rgb(172, 163, 64);
...
+    margin: 20px auto;
+    padding: 20px;
+    width: 80%;
+  }
+}
+
+input.dark-forms, select.dark-forms {
+  background-color: #fff;
+  border: #aaa 1px solid;
+  border-radius: 0 !important;
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/assets/default.scss]: I understand team has resolved to allow use of important.

File: src/views/Article/article.scss [hunk 11]
```diff
@@ -1,14 +1,16 @@
+@import '../../assets/default';
```
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/views/Article/article.scss:1]: I understand ESLint rules conflict for this. I will look at it.

**Review by olajide1234 — DISMISSED** [review]

**Review by codinger41 — DISMISSED** [review]

File: src/redux/reducers/authReducer.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (Dubby20) [reply src/redux/reducers/authReducer.js:1]: I did but eslint keeps throwing error.

File: src/__tests__/redux/actions/profile.test.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+      type: types.VIEW_PROFILE,
+      payload: profile,
+    };
+
+    expect(actions.getProfile(profile)).toEqual(expectedAction);
+  });
+});
+
+
+describe('handles getProfileRequest', () => {
```
Comment (9jaswag) [inline src/__tests__/redux/actions/profile.test.js]: @Dubby20 this is not a proper describe title

File: src/__tests__/redux/actions/profile.test.js [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+    const store = mockStore({ profile: {} });
+
+    return store.dispatch(actions.getProfileRequest()).then(() => {
+      expect(store.getActions()).toEqual(expectedActions);
+    });
+  });
+});
+
+
+describe('handles editProfileRequest', () => {
```
Comment (9jaswag) [inline src/__tests__/redux/actions/profile.test.js]: @Dubby20 same here

File: src/views/createArticle/createArticle.scss [hunk 14]
```diff
@@ -96,12 +95,14 @@ $black: #000;
...
   background-color: $white;
-  border: 2px solid $dark-col;
   border-radius: 8px;
   color: $dark-col;
   font-size: 20px;
   font-weight: bold;
-  padding: 8px 20px;
+  position: relative;
+  bottom: 16px;
+  // padding: 12px 20px;
```
Comment (9jaswag) [inline src/views/createArticle/createArticle.scss]: @Dubby20 please remove commented out code if not in use.

**Review by 9jaswag — CHANGES_REQUESTED** [review]
Was this feature demoed?

File: src/__tests__/redux/actions/profile.test.js (hunk 12 above)
Comment (Dubby20) [reply src/__tests__/redux/actions/profile.test.js]: Thanks. I will change it

File: src/views/createArticle/createArticle.scss (hunk 14 above)
Comment (Dubby20) [reply src/views/createArticle/createArticle.scss]: Okay. Thanks

File: src/__tests__/redux/actions/profile.test.js (hunk 13 above)
Comment (Dubby20) [reply src/__tests__/redux/actions/profile.test.js]: Okay

File: src/views/profilePage/profile.scss (hunk 9 above)
[the remark this answers is not in the document]
Comment (olajide1234) [reply src/views/profilePage/profile.scss]: I understand the team has resolved to allow the use of !Important.

**Review by codinger41 — DISMISSED** [review]

**Review by olajide1234 — DISMISSED** [review]

File: src/__tests__/views/viewProfile.test.js [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+import React from 'react';
+import { mount } from 'enzyme';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import ViewProfile from '../../views/profilePage/ViewProfile';
+
+
+describe('get a user profile', () => {
```
Comment (9jaswag) [inline src/__tests__/views/viewProfile.test.js]: @Dubby20 we're testing the `ViewProfile` component here, right? If so, please reword your description.

File: src/__tests__/views/editProfile.test.js [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+import React from 'react';
+import { mount } from 'enzyme';
+import { MemoryRouter } from 'react-router-dom';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import EditProfile from '../../views/profilePage/EditProfile';
+
+
+describe('edit a user profile', () => {
```
Comment (9jaswag) [inline src/__tests__/views/editProfile.test.js]: @Dubby20 same here...reword your description if we're testing a component

File: src/components/articleDescription/Index.jsx [hunk 17]
```diff
@@ -12,6 +13,6 @@ const ArticleDescription = (props) => {
 };
 
 ArticleDescription.propTypes = {
-  description: PropTypes.string.isRequired
+  // description: PropTypes.string.isRequired
```
Comment (9jaswag) [inline src/components/articleDescription/Index.jsx]: @Dubby20 please remove this if not in use

File: src/views/Article/Index.jsx [hunk 18]
```diff
@@ -51,21 +60,45 @@ const Article = (props) => {
...
+    const clientDocument = bodyElement.getBoundingClientRect();
+    const heightIsHtml = clientDocument.height;
+    const scrollMax = Math.ceil(heightIsHtml - innerHeight);
+    setProgress((scrollTop / scrollMax) * 100);
+  };
+
   useEffect(() => {
     getArticle(slug);
     getArticles();
+    window.addEventListener('scroll', handleScroll);
```
Comment (9jaswag) [inline src/views/Article/Index.jsx:77]: @Dubby20 what is this doing?

File: src/views/profilePage/EditProfile.jsx [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+
+
+  const onChangeImage = (event) => {
+    setImage(event.target.files[0]);
+    setImageUrl(URL.createObjectURL(event.target.files[0]));
+  };
+
+  const upload = async (event) => {
+    event.preventDefault();
+    const res = await props.uploadImage(image);
```
Comment (9jaswag) [inline src/views/profilePage/EditProfile.jsx]: @Dubby20 what's `res`?

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: src/components/articleDescription/Index.jsx (hunk 17 above)
Comment (Dubby20) [reply src/components/articleDescription/Index.jsx]: Alright. It was a mistake

File: src/views/Article/Index.jsx (hunk 18 above)
Comment (Dubby20) [reply src/views/Article/Index.jsx:77]: It updates the progress bar when scrolling the page

File: src/views/profilePage/EditProfile.jsx (hunk 19 above)
Comment (Dubby20) [reply src/views/profilePage/EditProfile.jsx]: It's a shorthand for response but I'll update it to have response instead of res

**Review by olajide1234 — APPROVED** [review]

**Review by kleva-j — APPROVED** [review]

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**